lib/command/src/Obelisk/Command/Project.hs

```diff
@@ -137,12 +138,29 @@ findProjectObeliskCommand target = do
         ]
       return Nothing
 
+-- We need the umask not be too strict with the group permissions on the folders
+-- Obelisk is using, since some systems have a default umask of 0002 instead of
+-- 0022, and we want to respect that. Since we have to set the umask to read it,
+-- because of http://man7.org/linux/man-pages/man2/umask.2.html#NOTES we have to
+-- be careful to call this operation in the initial single-threaded part of the
+-- program.
+
+-- | Get the umask for the Obelisk process.
+getDefaultUmask :: IO FileMode
+getDefaultUmask = do
+  initialMask <- setFileCreationMask safeUmask
+  void (setFileCreationMask initialMask)
+  pure initialMask
```

Comment:
I would feel better about this if we had a global `MVar` to make sure this is atomic.

Reply:
This might be one of the very few places where an *actual* global would be appropriate--because the race condition inherent to the system call itself.

```haskell
globalUmaskLock :: MVar ()
globalUmaskLock = unsafePerformIO (newMVar ())
{- NOINLINE globalUmaskLock -}
```

Reply:
Perfect, implemented